core/chains/evm/types/models.go

```diff
@@ -185,6 +195,18 @@ func (h *Head) ToInt() *big.Int {
 	return big.NewInt(h.Number)
 }
 
+func (h *Head) Equals(r *Head) bool {
+	if h == nil && r == nil {
+		return true
+	}
+
+	if h == nil || r == nil {
+		return false
+	}
+
+	return h.Hash == r.BlockHash() && h.Number == r.BlockNumber()
```

Comment:
Is this shorter form equivalent?
```suggestion
	return h.Hash == r.Hash && h.Number == r.Number
```

Reply:
Yup! thanks for pointing that out. Initally was going with a generic head type rather than pointer Head

Reply:
Yup! thanks for pointing that out. Initally was going with a generic head type rather than pointer Head